### PR TITLE
Rwa 750 workTypes cleanup

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-750

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by iguannago — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            get(ENDPOINT_PATH + "?filter-by-user=true").header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+        ).andReturn();
+
+        var expectedResponse = "{\"work_types\":[{\"id\":\"upper_tribunal\",\"label\":\"Upper Tribunal\"},"
+                               + "{\"id\":\"hearing_work\",\"label\":\"Hearing work\"}]}";
+        assertEquals(expectedResponse, postResponse.getResponse().getContentAsString());
+    }
+
+    @Test
+    void should_return_200_and_empty_list_when_work_types_are_given() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesControllerTest.java:106]: should this test name be adjusted? The workTypes are not given in this scenario

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTWorkTypeDatabaseServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    void should_retrieve_all_work_types() {
+
+        final List<WorkType> allWorkTypes = cftWorkTypeDatabaseService.getAllWorkTypes();
+
+        assertEquals(8, allWorkTypes.size());
+        assertEquals("hearing_work", allWorkTypes.get(0).getId());
+        assertEquals("Hearing work", allWorkTypes.get(0).getLabel());
+        assertEquals("upper_tribunal", allWorkTypes.get(1).getId());
+        assertEquals("Upper Tribunal", allWorkTypes.get(1).getLabel());
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTWorkTypeDatabaseServiceTest.java:37]: Nice. Could we include assertions for all the work types since they are finite?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+        ).andExpect(
+            ResultMatcher.matchAll(
+                status().isOk(),
+                jsonPath("$.work_types").isEmpty()
+            )
+        );
+    }
+
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesControllerTest.java:143]: what about scenario for getting all workTypes, i.e. /workTypes?

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesController.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            .body(new GetWorkTypesResponse<>(workTypes));
+
+    }
+
+    private Set<String> getActorWorkTypes(AccessControlResponse accessControlResponse) {
+        Set<String> roleWorkTypes = new HashSet<>();
+
+        for (RoleAssignment roleAssignment : accessControlResponse.getRoleAssignments()) {
+            String assignedWorkedList = roleAssignment.getAttributes().get(WORK_TYPES.value());
+            if (assignedWorkedList != null && !assignedWorkedList.isEmpty()) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesController.java:92]: could we add unit test to cover when assignedWorkedList is null or empty?

**Review by BenNzewi — COMMENTED** [review]
Did you notice the comment by Andy Wilkins (ExUI)? Not sure anyone has noticed. https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1511144970&__ncforminfo=x1HeSj6rOI9xK__T-uE_Z8Bn3cW6LziIZ1XAe0Bo24twB6yHrLny2eTPe9MSk86RlqWAXZR_FA-_Br6uxQ72zW6mcfpIQlIU
